lib/modules/manager/bun/artifacts.ts

```diff
@@ -15,15 +15,17 @@ export async function updateArtifacts(
 ): Promise<UpdateArtifactsResult[] | null> {
   const { packageFileName, updatedDeps, newPackageFileContent, config } =
     updateArtifact;
-  logger.debug({ updateArtifact }, `bun.updateArtifacts(${packageFileName})`);
+  logger.debug(`bun.updateArtifacts(${packageFileName})`);
   const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
 
   if (is.emptyArray(updatedDeps) && !isLockFileMaintenance) {
     logger.debug('No updated bun deps - returning null');
     return null;
   }
 
-  const lockFileName = config.lockFiles?.[0];
+  // Find the first bun dependency in order to handle mixed manager updates
+  const lockFileName = updatedDeps.find((dep) => dep.manager === 'bun')
+    ?.lockFiles?.[0];
```

Comment:
I would rather filter before calling updateArtifacts if we introduce a manager field.

Reply:
Maybe, but who knows if a manager will need to check for other manager updates in future, such as cocoapods and gems